### ISSUE TITLE
[feature request] [minor] load_state_dict specific option to not load buffers

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#33043]
I have buffers precomputed at model creation (in my case a precomputed FFT-like basis). Depending on whether this FFT frontend was part of the model object at train time, the related buffers may or may not be present in checkpoints.

However, if the FFT frontend is indeed in the model (e.g. in preparation of ONNX export), `load_state_dict` tries to load the buffers from checkpoint and fails if they're absent. I currently work around by `strict = False` (then it would also eat all legitimate errors about some old checkpoint format that may miss some conv weights), but maybe a more specific (separate?) option could be more expressive (`strict = 'ignore_buffers'` or `ignore_buffers = True` or `ignore_buffers = ['frontend.*'] or ignore_keys = ['frontend.*]'` or `ignore = lambda key: key.startswith('frontend')`)

**Comment by fmassa** [comment]
I think this might be a tricky thing as you mentioned -- in this case, it looks like you only want to keep some set of buffers, but not all of them (e.g., batch norm buffers should be loaded).

I think that for this particular case it might be better to let the user handle this themselves, as there can be plenty of corner cases depending on each specific use-case.

**Comment by fmassa** [comment]
I'm closing this issue, but let us know if you disagree and think that this can be made generic in a simple way.

**Comment by vadimkantorov** [comment]
@fmassa What do you think about the ignore_key lambda idea? I think it is generic enough and still allows the user to explicitly express the intent and provides an idiom that some model parameters/buffers should be disregarded without overriding load_state_dict

**Comment by vadimkantorov** [comment]
Stumbled on this again. Explicit `ignore_key` is less error-prone and more explicit than leaving `strict = False` and validating `missing_keys`, `unexpected_keys` manually
